client/asset/dcr/dcr.go

```diff
@@ -1138,20 +1181,30 @@ func (dcr *ExchangeWallet) getKeys(addr dcrutil.Address) (*secp256k1.PrivateKey,
 	return priv, pub, nil
 }
 
-// monitorConnection monitors the dcrwallet connection, sending an error via the
-// tipChange function when a disconnect is detected.
-func (dcr *ExchangeWallet) monitorConnection() {
-	ticker := time.NewTicker(time.Second * 10)
-out:
+// monitorBlocks pings for new blocks and runs the tipChange callback function
+// when the block changes.
+func (dcr *ExchangeWallet) monitorBlocks(ctx context.Context) {
+	h, _, err := dcr.node.GetBestBlock()
+	if err != nil {
+		dcr.tipChange(fmt.Errorf("error initializing best block for DCR: %v", err))
+	}
+	tipHash := *h
+	ticker := time.NewTicker(blockTicker)
+	defer ticker.Stop()
```

Comment:
Spent more time than I'd like to admit trying to figure out why we weren't receiving block updates on the client. Maybe this is expected behavior, but apparently `OnBlockConnected` doesn't come through when you connect through `dcrwallet`.

Reply:
That's certainly unexpected imo.  It may be a regression.